docs/plugins/customization.md

```diff
@@ -0,0 +1,69 @@
+---
+id: customization
+title: Customization
+description: Documentation on adding a customization logic to the plugin
+---
+
+## Overview
+
+The Backstage core logic provides a possibility to make the component customizable in such a way that the application
+developer can redefine the labels, icons, elements or even completely replace the component. It's up to each plugin
+to decide what can be customized.
+
+## For a plugin developer
+
+When you are creating your plugin, you have a possibility to use a metadata field and define there all
+customizable elements. For example
+
+```typescript jsx
+export type MyPluginMetadataProps = {
+  createButtonTitle: string;
+};
+
+export const myPluginMetadataRef = createMetadataRef<MyPluginMetadataProps>({
```

Comment:
Do you mean to do `MetadataProvider` as `EntityProvider` (where multiple of them), instead of as it is like `ApiProvider`, where it is one for the entire application.

Cause by having `useMetadata<MyMetadata>()` I can't keep it application-wise anymore, i need some key, to differentiate from which plugin it came, regardless will be it ref or just id as a string.

Reply:
Yep idd, intention would be to have plugin extensions wrapped with the context or something like that